tests/test_config.py

```diff
@@ -633,3 +635,48 @@ class Child(Parent):
         model_config: ConfigDict = {'str_to_lower': True}
 
     assert Child.model_config == {'extra': 'allow', 'str_to_lower': True}
+
+
+def test_replace_types():
+    LaxStr = Annotated[str, BeforeValidator(str_validator)]
+
+    class LaxStrBase(BaseModel):
+        model_config: ConfigDict = {'replace_types': {str: LaxStr}}
+
+    class Data(LaxStrBase):
+        f: str
+        i: str
+        d: str
+
+    user = Data(f=1.1, i=1, d=Decimal('1.1'))
+
+    assert user.model_dump() == {'f': '1.1', 'i': '1', 'd': '1.1'}
+
+
+def test_replace_types_nested():
+    LaxStr = Annotated[str, BeforeValidator(str_validator)]
+
+    class Inner(BaseModel):
+        x: str
+
+    class Outer(BaseModel):
+        x: str
+        inner: Inner
+        model_config = {'replace_types': {str: LaxStr}}
```

Comment:
Interesting. Is this the behavior for other config settings? That they don't bubble down? If so, perhaps add this to the test docstring.

Also, would be helpful to create a ticket for documenting this behavior as follow up work (just so this doesn't get lost in the mix).